calculator/main.py

```diff
@@ -25,9 +35,18 @@ def perform_operation(calc, operation, params):
     :param operation: String with the operation name. ie: 'add'
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
+    return: result of the operation
     """
-    pass
+    
+    if not all([isinstance(param, (int, float)) for param in params]):
+        raise InvalidParams("Given params are invalid.")
 
+    if operation in calc['operations'].keys():
```

Comment:
`keys()` is not necessary in this case because the `in` statement will look in there by default. Something like this should be enough:
```python
if operation in calc['operations']:
    ...
```